components/sp-solutions/org.wso2.carbon.sp.solutions.apim.analytics/widgets/APIMApiSubscriberAlertConfiguration/src/APIMApiSubscriberAlertConfiguration.jsx

```diff
@@ -0,0 +1,200 @@
+                legendTextColor: "#5d6e77"
+            },
+            pagination: true,
+            filterable: true,
+            append: false
+        };
+
+
+        this.metadata = {
+            "names": [
+                "applicationId",
+                "apiName",
+                "apiVersion",
+                "thresholdRequestCountPerMin"
+            ],
+            "types": [
+                "ordinal",
+                "ordinal",
+                "ordinal",
+                "ordinal"
+            ]
+        };
+         
+        this.state = {
+            width: this.props.glContainer.width,
+            height: this.props.glContainer.height,
+            data:[],
+            dataProviderConf:[],
+        };
+        
+        this.props.glContainer.on('resize', () =>
+            this.setState({
+                width: this.props.glContainer.width,
+                height: this.props.glContainer.height
+            })
+        );
+
+        this.handleDataReceived = this.handleDataReceived.bind(this);
+        this.handleDataManaged = this.handleDataManaged.bind(this);
+        this.handleSubmit = this.handleSubmit.bind(this);
+        this.handleChange = this.handleChange.bind(this);
+    }
+    
+    componentDidMount() {
+        super.getWidgetConfiguration(this.props.widgetID)
+            .then((message) => {
+            super.getWidgetChannelManager().subscribeWidget(this.props.id, this.handleDataReceived, message.data.configs.providerConfig);
+            this.setState({
+                    dataProviderConf : message.data.configs.providerConfig
+                }); 
+            })
+            .catch((error) => {
+                this.setState({
+                    faultyProviderConf: true
+                });
+            })        
+    }
+
+    componentWillUnmount(){
+        super.getWidgetChannelManager().unsubscribeWidget(this.props.id);
+    }
+
+    handleDataReceived(message) {
+        this.setState({
+            data:message.data
+        });
+    }
+
+    handleDataManaged(message){ 
+    }
+
+    handleChange(event){
+       this.setState({
+           [event.target.name]:event.target.value
+       });
+    }
+
+    handleSubmit(){
+       let dataProviderConfigs = _.cloneDeep(this.state.dataProviderConf);
+       let query;  
+       let applicationId = this.state.applicationId;
+       let apiName = this.state.apiName;
+       let apiVersion = this.state.apiVersion;
+       let tRequestCount = parseInt(this.state.tRequestCount);
+       
+       if(applicationId != null && apiName != null && apiVersion != null && tRequestCount != null){  
+            if(tRequestCount == 0){
+                query = dataProviderConfigs.configs.config.queryData.queryDelete;
+                query = query
+                .replace('{{applicationId}}',applicationId)
+                .replace('{{apiName}}',apiName)
+                .replace('{{apiVersion}}',apiVersion)
+            }
+            else{
```

Comment:
formatting error

Reply:
This was sample code added for revision and removed in 45ea197bf4dd881b1e9d79f5b3ba8d3cc12e05c9 in https://github.com/wso2/analytics-apim